backend/routes/partners.py

```diff
@@ -140,105 +149,292 @@ class Config:
+# user can join org they were invited to
+
+
+@bp.route("/join", methods=["POST"])
+@jwt_required()
+@min_role_required(UserRole.PUBLIC)
+def join_organization():
+    try:
+        body = request.get_json()
+        user_exists = PartnerMember.query.filter_by(
+            user_id=body["user_id"],
+            partner_id=body["partner_id"]).first()
+        if user_exists:
+            return {
+                "status" : "Error",
+                "message": "User already in the organization"
+            }, 400
+        else:
+            new_member = PartnerMember(
+                user_id=body["user_id"],
+                partner_id=body["partner_id"],
+                role=body["role"],
+                date_joined=datetime.now(),
+                is_active=True
+            )
+            db.session.add(new_member)
+            db.session.commit()
+            Invitation.query.filter_by(
+                user_id=body["user_id"],
+                partner_id=body["partner_id"]).delete()
+            db.session.commit()
+            return {
+                "status": "ok",
+                "message": "Successfully joined partner organization"
+            } , 200
+    except Exception:
+        db.session.rollback()
+        return {
+            "status": "Error",
+            "message": "Something went wrong!"
+        }, 500
+    finally:
+        db.session.close()
 
-    # TODO: Allow the API to accept a user email instad of a user id
-    # user_obj = User.get_by_email(request.context.json.user_email)
-    # if user_obj is None:
-    #     abort(400)
-
-    # new_member = PartnerMember(
-    #     partner_id=partner_id,
-    #     user_id=user_obj.id,
-    #     role=request.context.json.role,
-    # )
 
+# user can leave org they already joined
+@bp.route("/leave", methods=["DELETE"])
+@jwt_required()
+@min_role_required(UserRole.PUBLIC)
+def leave_organization():
+    """
+    remove from PartnerMember table
+    """
     try:
-        partner_member = partner_member_to_orm(request.context.json)
+        body = request.get_json()
+        result = PartnerMember.query.filter_by(
+            user_id=body["user_id"], partner_id=body["partner_id"]).delete()
+        db.session.commit()
+        if result > 0:
+            return {
+                "status": "ok",
+                "message": "Succesfully left organization"
+            }, 200
+        else:
+            return {
+                "status": "Error",
+                "message": "Not a member of this organization"
+            }, 400
     except Exception:
-        abort(400)
+        db.session.rollback()
+        return {
+            "status": "Error",
+            "message": "Something went wrong!"
+        }
+    finally:
+        db.session.close()
 
-    created = partner_member.create()
+# admin can remove any member from a partner organization
 
-    track_to_mp(
-        request,
-        "add_partner_member",
+
+@bp.route("/remove_member", methods=['DELETE'])
```

Comment:
Just like the role change, we shouldn't let an admin forcibly remove another admin without notification.